### ISSUE TITLE
App re-register fails

### ISSUE THREAD

**Issue by Midwaey** [BedrockLauncher/BedrockLauncher#39]
When attempting to launch any version of Minecraft Bedrock, I get the following "App re-register failed" Error. 

Refreshing or Restarting fixes nothing, the crash report is linked here:

```
Application Starting...
Application Pre-Initalization Finished!
Checking for updates
Current tag: 0.0.2.1
Latest tag: 0.0.2.2
New version available!
Removing package: Microsoft.MinecraftUWP_1.17.1004.0_x64__8wekyb3d8bbwe
Deployment progress: Processing 0%
Deployment progress: Processing 100%
Deployment done: Completed
Removal of package done: Microsoft.MinecraftUWP_1.17.1004.0_x64__8wekyb3d8bbwe
Registering package
Deployment failed: Another user has already installed a packaged version of this app. An unpackaged version cannot replace this. The conflicting package is Microsoft.MinecraftUWP and it was published by CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US.
App re-register failed:
System.Exception: Deployment failed: Another user has already installed a packaged version of this app. An unpackaged version cannot replace this. The conflicting package is Microsoft.MinecraftUWP and it was published by CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US.
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BedrockLauncher.ViewModels.GameManager.<DeploymentProgressWrapper>d__28.MoveNext() in D:\UserData\Source\mc_repos\BedrockLauncher\BedrockLauncher\ViewModels\GameManager.cs:line 596
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BedrockLauncher.ViewModels.GameManager.<ReRegisterPackage>d__27.MoveNext() in D:\UserData\Source\mc_repos\BedrockLauncher\BedrockLauncher\ViewModels\GameManager.cs:line 562
```

**Comment by quangtien07** [comment]
You can follow this tutorial to fix the issue: https://www.pmichaels.net/tag/the-current-user-has-already-installed-an-unpackaged-version-of-this-app/

**Comment by Midwaey** [comment]
I've gotten the correct paste link to the crash report: https://pastebin.com/raw/mh1W1tsj
Apologies for that.

**Comment by CoolAbhi1290** [comment]
@Midwaey can you paste the link in the original comment for ease of access

**Comment by Midwaey** [comment]
@CoolAbhi1290 Edited it in

**Comment by CarJem** [comment]
I encourage everyone with this issue to check out the latest beta with support for telling us exactly why this error is occurring so I can fix it!
https://github.com/BedrockLauncher/BedrockLauncher-Beta/releases/tag/0.0.2.3

**Comment by CoolAbhi1290** [comment]
I also started having this issue, but was able to fix it by reverting to `v0.0.1.8`.

Only this, and the older versions work for me because the newer versions suffer problems like not being able to download new versions because they cannot find a valid "download irl" and not being able to re register.

I believe the problem is with the launcher simply because it worked before and only happens when updating an installation like Latest Beta. Doesn't happen when you switch between versions.

**Comment by igoticecream** [comment]
when downgrading from latest beta to latest release:

```
System.Exception: Deployment failed: Windows cannot install package Microsoft.MinecraftUWP_1.17.1004.0_x64__8wekyb3d8bbwe because it has version 1.17.1004.0. A higher version 1.17.3020.0 of this package is already installed.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BedrockLauncher.ViewModels.GameManager.<DeploymentProgressWrapper>d__27.MoveNext() in D:\UserData\Source\mc_repos\BedrockLauncher\BedrockLauncher\ViewModels\GameManager.cs:line 591
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BedrockLauncher.ViewModels.GameManager.<ReRegisterPackage>d__26.MoveNext() in D:\UserData\Source\mc_repos\BedrockLauncher\BedrockLauncher\ViewModels\GameManager.cs:line 556
```

**Comment by THOBY555** [comment]
> You can follow this tutorial to fix the issue: https://www.pmichaels.net/tag/the-current-user-has-already-installed-an-unpackaged-version-of-this-app/

Its Not Working For Me Its Still The Same

**Comment by NT690** [comment]
Another workaround is to install MCMrARM's version launcher from here https://github.com/MCMrARM/mc-w10-version-launcher/releases/tag/0.2.1

Just download the versions you want and press the launch button. It may not be the best workaround, but it should work.

**Comment by CarJem** [comment]
I've been trying to address this issue... but if it's telling you a higher version already exists... your best bets are to manually uninstall the current version of bedrock without the launcher

**Comment by KanimeXP** [comment]
I got Different issue when launching all bedrock versions.
![Snímka obrazovky 2022-02-07 125434](https://user-images.githubusercontent.com/43906279/152783601-4c4295ca-1614-404a-bb29-57d7ba3f2c5f.png)

**Comment by KanimeXP** [comment]
Im using windows11 same issue on windows10

**Comment by KanimeXP** [comment]
Do someone know how to fix this?

**Comment by Rayth** [comment]
@KanimeXP that's issue #73

**Comment by Fabrico84** [comment]
i got a different problem while launching version
![Screenshot (27)](https://user-images.githubusercontent.com/104581143/165789632-1a1fe43b-ab72-4d17-881e-0f0c6bdefbc7.png)

**Comment by Rayth** [comment]
Try downloading the latest version. This error usually appears if you have a store installed minecraft installed so be sure to backup and uninstall that first.
https://github.com/BedrockLauncher/BedrockLauncher-Beta/releases/tag/2022.4.17.21

**Comment by Rayth** [comment]
Yes

**Comment by Fabrico84** [comment]
the newest version says this when I open the application:

_To run this application, you must install missing frameworks for .NET.
Would you like to download it now?
                                                                                Yes         No_

When I press yes it puts me to the website I used to download >NET 6.0.4 Runtime x64. What do I do?

**Comment by Fabrico84** [comment]
I followed them and now it opens but it crashes everytime i press next on where to save the files.

**Comment by Fabrico84** [comment]
I followed them and now it opens but till one point. After I press the make account button it crashes. Any ideas on how to fix that?

**Comment by TresAbhi** [comment]
@Fabrico84 Can you create a new issue with a log provided?

**Comment by Rayth** [comment]
Can anyone confirm if this issue still occurs in the 2022.4.17.21 beta linked in comments above?

**Comment by DelofJ** [comment]
Yes it still occurs
```
BedrockLauncher.Exceptions.PackageRegistrationFailedException: Deployment failed: Le package Microsoft.MinecraftUWP_1.17.1023.0_x64__8wekyb3d8bbwe avec l'origine Windows Store ne peut pas être chargé latéralement ou installé à l'aide de l'option du mode développeur.
 ---> System.Exception: Deployment failed: Le package Microsoft.MinecraftUWP_1.17.1023.0_x64__8wekyb3d8bbwe avec l'origine Windows Store ne peut pas être chargé latéralement ou installé à l'aide de l'option du mode développeur.
   at BedrockLauncher.Handlers.PackageHandler.DeploymentProgressWrapper(IAsyncOperationWithProgress`2 t) in D:\UserData\Source\visual_studio_repos\mc_repos\BedrockLauncher\BedrockLauncher\Handlers\PackageHandler.cs:line 473
   at BedrockLauncher.Handlers.PackageHandler.RegisterPackage(MCVersion v) in D:\UserData\Source\visual_studio_repos\mc_repos\BedrockLauncher\BedrockLauncher\Handlers\PackageHandler.cs:line 219
   --- End of inner exception stack trace ---
   at BedrockLauncher.Handlers.PackageHandler.RegisterPackage(MCVersion v) in D:\UserData\Source\visual_studio_repos\mc_repos\BedrockLauncher\BedrockLauncher\Handlers\PackageHandler.cs:line 230
   at BedrockLauncher.Handlers.PackageHandler.LaunchPackage(MCVersion v, String dirPath, Boolean KeepLauncherOpen) in D:\UserData\Source\visual_studio_repos\mc_repos\BedrockLauncher\BedrockLauncher\Handlers\PackageHandler.cs:line 62

```

**Comment by Rayth** [comment]
@DelofJ can you provide the full log file not just the exerpt that pops up when it occurs. If you're on the latest beta there is a "logs" folder. You'll find multiple files with date and timestamps in the name.

the D drive is just reference to where the app was developed nothing on your PC

**Comment by DelofJ** [comment]
I edited the previous comment to include the full crash log
Also, I noticed that when the crash happens and you want to click "View crash report" it wants to open a Log.txt file where the app is installed and doesn't open the latest log located in the `data` `logs` folder

**Comment by Rayth** [comment]
Ok can you try uninstall minecraft from control panel if it's there and then try running via the launcher?

**Comment by DelofJ** [comment]
I can't because the launcher has already uninstalled/unregistered Minecraft Bedrock from Windows
I tried to launch a version through the launcher i can launch, I uninstalled Minecraft Bedrock, and the same thing happened
The only thing I can do is install Minecraft with the Appx I saved
Also, I'm using the Launcher [Beta 2022.4.17.21](https://github.com/BedrockLauncher/BedrockLauncher-Beta/releases/tag/2022.4.17.21)

**Comment by Rayth** [comment]
Ok final test I can think of, install minecraft from the store, run it once then uninstall it, then try installing and launching from the launcher.

**Comment by Rayth** [comment]
Ok that could be something with how we had to change the code to work with previews then and most likely won't get changed again due to betas being discontinued. Features that were in old betas (in your case 1.17.10) would have been in release version1.17.10 and 1.17.11 and later

**Comment by Rayth** [comment]
> @Rayth This isn't solved in 2022.7.3.62

We need a fresh report for the new version of the launcher, complete with the new Log file (installation folder > data > logs)
